### PR TITLE
impl: threshold encryption and decryption [ISSUE-81]

### PR DESCRIPTION
**Pull request by boohyunsik**
T P K E

### CODE REVIEW

File: threshold_encrpytion.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package cleisthenes
+
+import "github.com/boohyunsik/tpke"
```
Comment (AgwaB) [inline threshold_encrpytion.go]: we have DE-labtory/tpke!

File: threshold_encrpytion.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	secretKeySet := tpke.RandomSecretKeySet(c.threshold)
+	publicKeySet := secretKeySet.PublicKeySet()
+	return &KeySet{
+		pkSet: publicKeySet,
+		skSet: secretKeySet,
+	}
+}
+
+// Encrypt encrypts some byte array message.
+// Length of byte array must be shorter than 32 bytes. (TODO : fix it.)
```
Comment (AgwaB) [inline threshold_encrpytion.go]: Does `TODO` plan to fix the `Length of byte array must be shorter than 32 bytes.`?

File: threshold_encryption_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+func TestSetUp(t *testing.T) {
+	config := Config{
+		threshold:   3,
+		participant: 6,
+	}
+
+	var privPubKeySet *tpke.PublicKeySet
+	var privSecKeySet *tpke.SecretKeySet
+
+	for i := 0; i < 100; i++ {
```
Comment (AgwaB) [inline threshold_encryption_test.go]: How about compare as `set`?
```go
privPubKeySet := make(map[*tpke.PublicKeySet]bool)
privSecKeySet := make(map[*tpke.SecretKeySet]bool)
for i := 0 ; i < 100 ; i++ {
       keySet := Setup(config)
       privPubKeySet[keySet.pkSet] = true
       privSecKeySet[keySet.skset] = true
}

if len(privPubKeySet) != 100 || len(privSecKeySet) != 100 {
    error
}
```

File: threshold_encryption_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+func TestEncrypt(t *testing.T) {
+	config := Config{
+		threshold:   3,
+		participant: 6,
+	}
+
+	msg := []byte("honeyBadger BFT")
+	var privCipherText *tpke.CipherText
+	keySet := Setup(config)
+	for i := 0; i < 100; i++ {
```
Comment (AgwaB) [inline threshold_encryption_test.go]: Same as above.

File: threshold_encryption_test.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/boohyunsik/tpke"
+)
+
+func TestSetUp(t *testing.T) {
+	config := Config{
+		threshold:   3,
+		participant: 6,
+	}
+
+	var privPubKeySet *tpke.PublicKeySet
```
Comment (zeroFruit) [inline threshold_encryption_test.go]: plz use `cleisthenes.PublicKeySet`

File: threshold_encryption_test.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
+package cleisthenes
```
Comment (zeroFruit) [inline threshold_encryption_test.go]: change package name with `cleisthenes_test` plz

**Review by zeroFruit — APPROVED** [review]
nice work!

**Pull request merged**